api/app/Actions/Upload/UploadImageAction.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Upload;
+
+use App\Services\ImageUploader;
+
+final class UploadImageAction
+{
+    private ImageUploader $imageUploader;
+
+    public function __construct(ImageUploader $imageUploader)
+    {
+        $this->imageUploader = $imageUploader;
+    }
+
+    public function execute(UploadImageRequest $request): UploadImageResponse
+    {
+        $result = $this->imageUploader->upload($request->getFile(), $request->getUserId(), $request->getType());
+
+        if ($request->getOldFile()) {
+            $this->imageUploader->remove($request->getOldFile());
+        }
+
+        return new UploadImageResponse($result);
```

Comment:
First of all we use it in tests  Storage::disk('local')->assertExists($content['path']); Also maybe better to store the storage path in DB instead of URL in case we switch the storage we should only regenerate URL not alter DB records.

Reply:
@aussie-m-mike The path can be parsed from the URL. You just need to get the base path of the storage and exclude it from the URL.